### ISSUE TITLE
Unable to solve nonlinear system with NewtonSolver.

### ISSUE THREAD

**Issue by ArnoiZ** [OpenTidalFarm/OpenTidalFarm#32]
Hi,

I have been trying to perform tidal simulation with OpenTidalFarm and have created all the input data it requires by now. I followed the example tidal-simulation for orkney and changed input data to the ones I created. But I got the following result.
```
*** Error:   Unable to solve nonlinear system with NewtonSolver.
*** Reason:  Newton solver did not converge because maximum number of iterations reached.
*** Where:   This error was encountered inside NewtonSolver.cpp.
*** Process: 0
*** 
*** DOLFIN version: 2017.1.0
*** Git changeset:  3d1f687ec9ee39afc0fe6e01800431995b42ad04
*** -------------------------------------------------------------------------
```

I have also tried smaller timesteps, but the result is still the same.

I have packed everything I used in [this package](https://drive.google.com/open?id=1sVMUwsx89ghpT_7Y4EXJ_nzElskmYqOF) (Because the package is bigger than 10 MB, I had to upload it to google drive), could you help me take a look where the problem is?

**Comment by funsim** [comment]
Hi,

This error tells you that the non-linear solver for the Navier-Stokes equations did not converge for one of the timesteps. The exact reason for this is unclear from this message. 

My advice is that you try to simplify the setup to get the simulation to work. A good start would be for instance to set the bathymetry to a constant depth (maybe around 100m).

**Comment by ArnoiZ** [comment]
Hi,

I just set `z` variable in bathymetry to 100 and rerun the simulation, but this problem still happens.
Is there something else I could try to identify where the problem is?

**Comment by funsim** [comment]
The next typical issue are boundary conditions or initial conditions or potential issues with the mesh. If the simulation starts to fail after a few timesteps, then you can open the vtk file and look at the pressure and velocity before and see if you see anything suspicious.

**Comment by ArnoiZ** [comment]
Last time I asked whether `*_physical_region.xml` is necessary if only tidal simulation is required and you said it didn't. 
I just found out that because I made a mistake when creating geo file by naming physical surface '1', _physical _region.xml was generated, and I think this might be the reason causing the problem.
So I just changed the physical surface index to zero and _physical_region.xml is no longer generated.

However, another problem appeared.
When using convert_to_new_xml.py, it returns with an error saying that cannot open `_physical_region.xml` while it should be skipping it according to the script. 

It also happens when I tried to follow 
`For stability reasons, we want to increase the viscosity at the inflow and outflow boundary conditions. For that, we read in a precomputed function (generated by 'compute_distance')`
and use compute_distance.py to generate `dist.xml`
```
*** Error:   Unable to read data from XML file. 
*** Reason:  Unable to open file "edoshima_physical_region.xml".
```
Is there a work around for this problem?

**Comment by funsim** [comment]
The setup files that you posted at the beginning look fine. The code actually runs if I use the following boundary conditions:

```
bcs = BoundaryConditionSet()
bcs.add_bc("eta", "1.1", facet_id=1)
bcs.add_bc("eta", "0.9", facet_id=2)
```
That is, if I use constant bcs instead of the `TidalForcing`  class. This indicates that your tidal boundary conditions have an issue. This is confirmed if you replace the timeloop at the end with the following code:
```

X = FunctionSpace(domain.mesh, "CG", 1)
x = Function(X)
facet_ids = domain.facet_ids

for _, expr, facet_id, _ in bcs.filter("eta", "strong_dirichlet"):
    bc = DirichletBC(X, expr, facet_ids, facet_id)
    bc.apply(x.vector())
print("Eta dirichlet BC values:")
print(x.vector().array())
```
You will see `nan` values in the resulting vector. 

In summary, you need to fix your `TidalForcing` setup. In particular check that all the parameters are set correctly (see also the documentation in `https://github.com/stephankramer/uptide/blob/master/uptide/tidal_netcdf.py`)

**Comment by ArnoiZ** [comment]
How should I remedy the problem that I could not calculate `dist.xml` and load the mesh because I don't have '_physical_region.xml'? 
Should I just add one farm marker to the geo file?

**Comment by funsim** [comment]
I checked the computed distance function in your original zip file (in that setup the _physical_region.xml exists), and the result looks just fine.

**Comment by ArnoiZ** [comment]
But I thought the xml file was wrong because I didn't add any physical region for farm markers but `_physical_region.xml` was generated anyway.

Here's the physical surface in the original geo file:
`Physical Surface(1) = {1};`
I compared my geo file with the example and noticed that physical surface in example geo file used 1 for the physical region. I think that why `_physical_region.xml` was generated, so I changed it to zero: 
`Physical Surface(0) = {1};`,
and `_physical_region.xml` is no longer generated, which caused the problem I mentioned earlier.

Here's the mesh I generated with the changed geo file. And it could not pass `compute_distance.py` nor load mesh.
[mesh.zip](https://github.com/OpenTidalFarm/OpenTidalFarm/files/3519330/mesh.zip)

**Comment by funsim** [comment]
If I remember correctly (haven't looked at the code for some time), the `_physical_region.xml` is loaded (and assumed that the file exists), but it is not actually used in a pure simulation like the one you are running.

**Comment by ArnoiZ** [comment]
Okay, maybe I’ll try to add a dummy farm marker to generate
_physical_region.xml and see if it works, thank you very much.

令和元年8月20日(火) 17:32 Simon Wolfgang Funke <notifications@github.com>:

**Comment by ArnoiZ** [comment]
Hi,

I have fixed the mesh and added a dummy farm marker to generate _physical_region.xml, but the simulation still could not run.
You mentioned that there could be an issue with tidal boundary conditions, do you mean the problem is with the mesh?

Could you tell me how could I remedy the issue and which parameters should I set properly to get the simulation going?
